numpy/core/numeric.py

```diff
@@ -2250,12 +2260,13 @@ def allclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
     return bool(res)
 
 
-def _isclose_dispatcher(a, b, rtol=None, atol=None, equal_nan=None):
+def _isclose_dispatcher(a, b, rtol=None, atol=None, equal_nan=None,
+                        casting=None):
     return (a, b)
 
 
 @array_function_dispatch(_isclose_dispatcher)
-def isclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
+def isclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False, casting='unsafe'):
```

Comment:
```suggestion
def isclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False, casting='same_kind'):
```
With my initial `can_cast` suggestion the default can now be set back to `same_kind`, 
_i.e._ the same (implicit) value that is used in the current master via the ufuncs.